server/paymentProviders/stripe/webhook.ts

```diff
@@ -92,6 +92,35 @@ async function createOrUpdateOrderStripePaymentMethod(
   return pm;
 }
 
+export const mandateUpdated = async (event: Stripe.Event) => {
+  const stripeMandate = event.data.object as Stripe.Mandate;
+
+  const stripePaymentMethodId =
+    typeof stripeMandate.payment_method === 'string' ? stripeMandate.payment_method : stripeMandate.payment_method.id;
+
+  const paymentMethod = await models.PaymentMethod.findOne({
+    where: {
+      data: {
+        stripePaymentMethodId,
+      },
+    },
+  });
+
+  if (!paymentMethod) {
+    return;
+  }
+
+  await paymentMethod.update({
+    saved: stripeMandate.type === 'multi_use' && stripeMandate.status !== 'inactive',
+    data: {
+      ...paymentMethod.data,
+      stripeMandate,
+    },
+  });
+
```

Comment:
Wouldn't be a good idea to cancel any subscriptions if the mandate goes inactive?

Reply:
Stripe mentions that the [Bacs mandate](https://stripe.com/docs/payments/payment-methods/bacs-debit#mandates) can be reactivated, I think we can check this at time of subscription charge to give it time for a reactivation / change of payment method.